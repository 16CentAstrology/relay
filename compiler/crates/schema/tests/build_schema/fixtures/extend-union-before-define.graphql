# expected-to-throw

schema {
  query: Query
}

type Query {
  foo: MyType
}

type MyType {
  id: ID!
}

# Invalid accoring to spec:
# > The named type must already be defined and must be an Union type.
# -- https://spec.graphql.org/June2018/#InterfaceTypeExtension
extend union MyUnion = MyType

union MyUnion
