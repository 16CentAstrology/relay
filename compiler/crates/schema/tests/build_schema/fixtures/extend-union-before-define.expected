==================================== INPUT ====================================
# expected-to-throw

schema {
  query: Query
}

type Query {
  foo: MyType
}

type MyType {
  id: ID!
}

# Invalid accoring to spec:
# > The named type must already be defined and must be an Union type.
# -- https://spec.graphql.org/June2018/#InterfaceTypeExtension
extend union MyUnion = MyType

union MyUnion
==================================== ERROR ====================================
✖︎ Cannot extend type 'MyUnion', the type is not defined on the server schema.

  extend-union-before-define.graphql:18:14
   17 │ # -- https://spec.graphql.org/June2018/#InterfaceTypeExtension
   18 │ extend union MyUnion = MyType
      │              ^^^^^^^
   19 │
